_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -1234,47 +1236,60 @@ mfxStatus VAAPIVideoCORE::IsGuidSupported(const GUID /*guid*/,
     if (IsMVCProfile(par->mfx.CodecProfile))
         return MFX_WRN_PARTIAL_ACCELERATION;
 
-    switch (par->mfx.CodecId)
+    MFX_CHECK(m_Display, MFX_ERR_DEVICE_FAILED);
+
+    VaGuidMapper mapper(guid);
+    VAProfile req_profile = mapper.profile;
+    VAEntrypoint req_entrypoint = mapper.entrypoint;
+    mfxI32 va_max_num_entrypoints = vaMaxNumEntrypoints(m_Display);
+    mfxI32 va_max_num_profiles = vaMaxNumProfiles(m_Display);
+    MFX_CHECK(va_max_num_entrypoints & va_max_num_profiles, MFX_ERR_UNSUPPORTED);
```

Comment:
why bitwise "and"? Maybe you wanted "&&"

Reply:
No. I will write ||, because this is more readable.

Reply:
Did you mean && ? Logical or makes no sense here.